src/Components/Components/src/Rendering/ComponentState.cs

```diff
@@ -53,6 +53,15 @@ public ComponentState(Renderer renderer, int componentId, IComponent component,
         public IComponent Component { get; }
         public ComponentState ParentComponentState { get; }
         public RenderTreeBuilder CurrentRenderTree { get; private set; }
+        public bool IsRootComponent { get; set; }
+
+
+        /// <summary>
+        /// A snapshot of the <see cref="ParameterView"/> used to initially render a component.
+        /// This is used to capture parameters specified via <see cref="Renderer.RenderRootComponentAsync(int, ParameterView)"/>
+        /// when hot reload is enabled.
+        /// </summary>
+        public IReadOnlyDictionary<string, object>? InitialParameters { get; set; }
```

Comment:
Is this value being updated before every hot reload is triggered or is it just saved once?

Reply:
Currently there are no APIs for changing the parameters on a root component, so we should be OK for now at least :)